api/src/classes/company.py

```diff
@@ -0,0 +1,156 @@
+# pyright: reportOptionalMemberAccess=false
+
+from typing import TYPE_CHECKING, Any
+
+from fastapi import HTTPException
+from sqlalchemy.exc import SQLAlchemyError
+from sqlmodel import Session, desc, or_, select
+from src.classes.pagination import CompanyPagination, Paginate
+from src.models import Company, CompanyCreate, CompanyPublic, CompanyUpdate
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+
+class CompanyRepresentation:
+    """Class that represents a `company` that is stored in the database."""
+
+    def __init__(self, session: Session, *, company: Company | None = None) -> None:
+        self.session: Session = session
+        self.company: Company | None = company
+
+    @classmethod
+    def fetch_company(
+        cls, session: Session, *, name: str | None = None, owner_id: str | None = None, company_id: int | None = None
+    ) -> "CompanyRepresentation":
+        """Return instance with target Company, if it exists.
+
+        :param company_id: ID of the Company to search for.
+        :param session: Database session.
+        :param name: Name of the company.
+        :param owner_id: ID of the Owner.
+        :return: Instance with the fetched company.
+        """
+        fetched_company: Company | None = session.exec(
+            select(Company).where(
+                or_(Company.id.__eq__(company_id), Company.name.__eq__(name), Company.owner_id.__eq__(owner_id))
```

Comment:
> Instead of using the `__eq__` dunder, SQLModel supports normal python expressions for equality (see: https://sqlmodel.tiangolo.com/tutorial/where/#filter-rows-using-where-with-sqlmodel)

Ya, sorry. From a seperate project I did, mainly with FastAPI and SQLalchemy, using normal python expressions as mentioned was causing the API (and IDE) to be unhappy. 

Will make those changes